File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day02Test.java
package com.capital7software.aoc.aoc2023.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day02Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day02Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day02();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void getSumOfGameResultIds() {
    var instance = new Day02();
    int expected = 8;
    var actual = instance.getSumOfGameResultIds(lines);
    assertEquals(expected, actual, "The sum of possible Game IDs "
        + "is not what was expected: " + expected);
  }

  @Test
  void getSumOfGameResultPower() {
    var instance = new Day02();
    int expected = 2286;
    var actual = instance.getSumOfGameResultPower(lines);
    assertEquals(expected, actual, "The sum of Game Powers "
        + "is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2018/src/test/kotlin/com/capital7software/aoc/aoc2018aoc/days/Day02Test.kt
package com.capital7software.aoc.aoc2018aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day02Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day02Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day02()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testGetChecksum() {
    val instance = Day02()
    val expected = 12
    val actual = instance.getChecksum(lines)
    Assertions.assertEquals(expected, actual, "The checksum "
        + "is not what was expected: $expected")
  }

  @Test
  fun testGetPrototypeBoxIds() {
    setupFromFile("inputs/input_day_02-02.txt")
    val instance = Day02()
    val expected = Triple("fghij", "fguij", "fgij")
    val actual = instance.getPrototypeBoxIds(lines)
    Assertions.assertEquals(expected, actual, "The prototype boxes "
        + "is not what was expected: $expected")
  }
}
